cmd/scollector/collectors/continuous.go

```diff
@@ -0,0 +1,95 @@
+package collectors
+
+import (
+	"reflect"
+	"regexp"
+	"runtime"
+	"time"
+
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+type ContinuousCollector struct {
+	F    func(collectorStatsChan chan<- *ContinuousCollectorStats)
+	name string
+	tags opentsdb.TagSet
+	init func()
+}
+
+type ContinuousCollectorStats struct {
+	duration float64
+	result   int
+}
+
+var ContinuousCollectorVars struct {
+	dpChan                chan<- *opentsdb.DataPoint
+	quit                  <-chan struct{}
+	reTwoOrMoreUnderscore *regexp.Regexp
+	nameWorkerChan        map[string]chan *ContinuousCollectorStats
+}
+
+func init() {
+	ContinuousCollectorVars.reTwoOrMoreUnderscore = regexp.MustCompile("[_]{2,}")
+	ContinuousCollectorVars.nameWorkerChan = make(map[string]chan *ContinuousCollectorStats)
+}
+
+func (c *ContinuousCollector) Init() {
+	if c.init != nil {
+		c.init()
+	}
+}
+
+func (c *ContinuousCollector) Run(dpChan chan<- *opentsdb.DataPoint, quit <-chan struct{}) {
```

Comment:
It wasn't that the interval collector could not be made to work, it was it did not seem ideal.

If you had a query to run every 15 seconds and other query run every 20 seconds, how often would the interval collector have to be set to run? My understanding is you would have to have interval collector run every 5 seconds otherwise the timing would be off. If you had even more queries with different timings the interval collector would have to be set to every second to check if there is a query to be run.

Also what if had a query that ran every 20 seconds and another query that ran every 30 minutes that took 5 minutes to run. Would you run them in the same interval collector? If you did, would you run both the queries at the same time? If you did run them at the same time, the MultiDataPoint could not be returned until both were finished.

It just seems to me a better solution would be to have a continuous collector, just like the external files have.